File: BackEnd/backend/src/main/java/com/example/backend/entities/content/Content.java
package com.example.backend.entities.content;

import com.example.backend.entities.poi.PointOfInterest;
import com.example.backend.entities.users.User;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.Accessors;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.util.Date;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@Entity
@Table(name = "content")
public class Content {

    @Id
    @Column(name = "content_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Lob
    private String content;

    private String hashtag;

    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    private Status status;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private Date createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private Date updatedAt;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User author;

    @ManyToOne
    @JoinColumn(name = "poi_id")
    private PointOfInterest poi;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "file_id")
    private MediaFile mediaFile;
}

File: Frontend/src/app/core/interfaces/content.ts
import { MediaFile } from "./media.file";
import { PointOfInterest } from "./point.of.interest";
import { User } from "./user";

export interface Content {
    id: number,
    title: string,
    content: string,
    hashtag: string,
    author: User,
    status: Status,
    pointOfInterest: PointOfInterest,
    mediaFile?: MediaFile,
    createdAt: Date,
    updatedAt: Date,
}


export enum Status {
    PENDING,
    APPROVED,
    REJECTED
}
